src/client.ts

```diff
@@ -1678,6 +1683,22 @@ export class Client<Ctx = null> {
       this.channels = {};
       if (closeResult.closeReason !== ClientCloseReason.Error) {
         // if we got here as a result of an error we're not gonna call onUnrecoverableError again
+        // but we are going to dump a bunch of breadcrumbs to help chase down how we got here.
+
+        for (const key in this.channels) {
+          const channel = this.channels[key];
```

Comment:
nit 

```suggestion
        for (const channel in Object.values(this.channels)) {
```

also avoids https://eslint.org/docs/latest/rules/guard-for-in

Reply:
OK, sort of done.